tests/common/minio.rs

```diff
@@ -0,0 +1,41 @@
+use std::env;
+use std::process::ExitStatus;
+
+use anyhow::{Context, Result};
+use tokio::process::Command;
+use tokio::sync::oneshot;
+use tokio::task::JoinHandle;
+use url::Url;
+
+use super::{free_port, spawn_server};
+
+/// Spawn a minio server to use for testing
+pub async fn start_minio() -> Result<(
```

Comment:
Does this now mean minio is a requirement locally for running `cargo nextest run`?

Reply:
Yeah, so I'm not sure exactly how we want to handle this, but similar to `redis` and `vault`. 

CI should be fine, as the nix stuff has been updated, but people do have to have this installed locally to run the tests

Reply:
I could move these tests to under a specific feature flag though, if `minio` is one too far

Reply:
I would be okay if the result is that that individual test fails, but doesn't bring down the entire test suite

Reply:
So that's *kind of* what happens -- the whole test suite doesn't go down -- only the part of the suite that does minio stuff which is `tests/blobstore-s3.rs` for now

Reply:
yeah unlike before/with the other `Makefile` driven stuff, the dependencies are started *during* the test. 

We could reduce our dependencies to one if we... bring in the :whale2: ... Which has been discussed before but people weren't so onboard with it